cuqipy_fenics/utilities.py

```diff
@@ -1,6 +1,62 @@
+import numpy as np
 import dolfin as dl
+import warnings
 from numbers import Number
 
+
+def compute_stats(samples):
+    """This function computes the statistics (mean and variance) of a set of 
+    samples on the function value representation. Two different approaches 
+    used in computing the variance: The first approach is computing the variance
+    var1 using the FEniCS functions directly, and is general for any FEM 
+    function. The second approach is computing the variance var2 using the
+    FEniCS vectors directly, and is specific for some FEM function spaces,
+    e.g. CG1. The returned values are the mean, var1, and var2."""
+
+    geom = samples.geometry
+    V = geom.function_space
+
+    # Loop to compute the samples function value
+    sample_funs = []
+    sample_funs_dof_vecs = np.empty((V.dim(), samples.samples.shape[1]))   
```

Comment:
`samples.samples.shape[-1]` might be more robust

Reply:
👍